### PR TITLE
Add checksum for py-numexprt 2.8.3

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-numexpr/package.py [hunk 1]
```diff
@@ -21,8 +22,10 @@ class PyNumexpr(PythonPackage):
...
     version("2.4.6", sha256="2681faf55a3f19ba4424cc3d6f0a10610ebd49f029f8453f0ba64dd5c0fe4e0f")
 
-    depends_on("python@2.7:", when="@2.7.3:", type=("build", "run"))
+    depends_on("python@3.7:", when="@2.8.3:", type=("build", "run"))
+    depends_on("python@2.7:", when="@2.7.3", type=("build", "run"))
     depends_on("python@2.6:", when="@:2.7.2", type=("build", "run"))
     # https://github.com/pydata/numexpr/issues/397
-    depends_on("py-numpy@1.7:1.22", type=("build", "run"))
+    depends_on("py-numpy@1.7:1.22", type=("build", "run"), when="@:2.7")
+    depends_on("py-numpy@1.13.3:", type=("build", "run"), when="@2.8.3:")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-numexpr/package.py:30]: Missing a dependency on `py-packaging`

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-numexpr/package.py [hunk 2]
```diff
@@ -21,8 +22,11 @@ class PyNumexpr(PythonPackage):
...
     depends_on("py-setuptools", type="build")
+    depends_on("py-packaging", type="build", when="@2.8.3:")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-numexpr/package.py]:
```suggestion
    depends_on("py-packaging", type=("build", "run"), when="@2.8.3:")
```

**Review by adamjstewart — DISMISSED** [review]

**Review by adamjstewart — DISMISSED** [review]

**Comment by iarspider** [comment]
@adamjstewart the pipeline failure is not related to this PR: it is an issue with the combination of numpy and intel compiler https://github.com/numpy/numpy/issues/22011

**Comment by adamjstewart** [comment]
Hmm, not sure what could have changed in the pipeline where that would start to break all of a sudden. @scottwittenburg?

**Comment by scottwittenburg** [comment]
> Hmm, not sure what could have changed in the pipeline where that would start to break all of a sudden

Am I just reading the diff wrong, or didn't this PR explicitly change the version of numpy getting built in the pipeline?

**Comment by adamjstewart** [comment]
Oh, that could be an unintentional consequence of this PR, yes.

**Comment by iarspider** [comment]
@adamjstewart @scottwittenburg should I add a `conflict(...)` clause to py-numpy recipe?

**Comment by adamjstewart** [comment]
Yes, I think that's the right solution. @rgommers?

**Comment by rgommers** [comment]
Yes, makes sense to me. NumPy 1.23.0 shipped moved some code from C to C++ that the Intel Compiler seems to choke on, so until that is resolved there's a conflict there.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**